src/main.ts

```diff
@@ -174,14 +175,42 @@ export class AdViewManager {
 	private fetch: any
 	private options: AdViewManagerOptions
 	private timesShown: { [key: string]: number }
+	private optsUpdated: boolean
 	private getTimesShown(channelId: string): number {
 		return this.timesShown[channelId] || 0
 	}
 	constructor(fetch, opts: AdViewManagerOptions) {
 		this.fetch = fetch
 		this.options = { ...defaultOpts, ...opts }
 		this.timesShown = {}
+		this.optsUpdated = false
 	}
+	async applyMarketOptions() {
+		const {
```

Comment:
can't you just do `const opts = this.options` and then just use `opts.*`? It would be cleaner IMO